core/src/main/java/com/amplifyframework/auth/AuthUser.java

```diff
@@ -52,6 +55,24 @@ public String getUsername() {
         return username;
     }
 
+    /**
+     * Interface for changing the password of an existing user.
+     * @param oldPassword The user's existing password
+     * @param newPassword The new password desired on the user account
+     * @param onSuccess Success callback
+     * @param onError Error callback
+     */
+    public abstract void changePassword(
```

Comment:
If `AuthUser` is a valid working user, and then `changePassword(...)` is called and fails, then `AuthUser` is no longer a valid working user. This might have significance to the auth system, and to the app that integrates it. How can other components come to know that that `AuthUser` no longer has a valid password? Only by attempting to use the instance?

This also makes it look like there's one auth, and it mutates. Is that the case? If so, should you make this a singleton? If so, you could end up with an access pattern like `Amplify.Auth.User.changePassword(...)` which is certainly very _readable_, anyway.

Instead of making an `abstract` class, and allowing an implementer to add a variety of customizations onto this class, you could accomodate the `changePassword(...)` by accepting a `ChangePasswordStrategy` into the constructor of the `final` `AuthUser`. e.g.,

```
@FunctionalInterface interface ChangePasswordStrategy {
    void changePassword(String oldPass, String newPass, Action onSuccess, Consumer<AuthException> onFailure);
}
```

Reply:
Let's talk about this on Chime - I'm not fully understanding the concerns you raise.